app/questionnaire/path_finder.py

```diff
@@ -178,3 +200,23 @@ def _remove_rule_answers(self, goto_rule, this_location):
                     self.answer_store.remove_answer(condition["id"])
 
         self.progress_store.remove_completed_location(location=this_location)
+
+
+def should_goto_new(
+    rule, schema, answer_store, list_store, metadata, location, routing_path_block_ids
+):
+    when_rule_evaluator = WhenRuleEvaluator(
+        schema,
+        answer_store,
+        list_store,
+        metadata,
+        location,
+        routing_path_block_ids,
+    )
```

Comment:
The `WhenRuleEvaluator` should be initialised once in the init method.

Reply:
I moved it to top level of `_evaluate_routing_rules` method because `__init__` of PathFinder doesn't have access to arguments like `location` and `routing_path_block_ids`. Consulted this with @MebinAbraham, he said to keep it simple for now.